### PR TITLE
API Testing | Mocking and Logger docs

### PR DESCRIPTION
**Pull request by kimsauce**
### Description

See https://saucedev.atlassian.net/browse/TCD-1989

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation fix (typos, incorrect content, missing content, etc.)

### Checklist

- [x] I have read the [contributing](https://github.com/saucelabs/sauce-docs/blob/main/CONTRIBUTING.MD) document.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: docs/api-testing/mark3/mocking.md [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+Why the name Piestry?
+didn't want to drop the Sauce tradition of naming stuff after food, so Piestry is:
+"A pastry that acts as if it was a pie, but is nothing more than a pastry"
+
+
+--use cases: you want to fake the payment transaction, you want to isolate the microservice from the rest of the API actions (ie everything else is stable and you want to drill down to find the error)
+
+
+Where can I run it?
+* Docker container (locally or CI/CI pipeline)
```
Comment (LoryApiFortress) [inline docs/api-testing/mark3/mocking.md:26]: CI/CD pipeline

Comment (LoryApiFortress) [inline docs/api-testing/mark3/mocking.md:26]: CI/CD pipeline

File: docs/api-testing/mark3/mocking.md [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+  * Image: quay.io/saucelabs/piestry
+  * Run it using the code snippet below:
+  ```bash
+  docker run -v "$(pwd)/specs:/specs" -p 5000:5000 quay.io/saucelabs/piestry -u /specs/myspec.yaml
+  Where /specs/myspec.yaml is the URI to a YAML file, local or remote.
```
Comment (LoryApiFortress) [inline docs/api-testing/mark3/mocking.md:31]: where $(pwd)/myspec means the {current_directory}/myspec that gets mounted to the container in the /specs folder.
Therefore, the -u (relative to the container is) /specs/myspec.yaml

File: docs/api-testing/mark3/mocking.md [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+  2021-10-05T07:32:35.157Z info: Piestry booting on port: 5000
+  2021-10-05T07:32:35.189Z info: Registering GET /api/v1/release-notes
+  2021-10-05T07:32:35.191Z info: Registering GET /api/v1/user
+  2021-10-05T07:32:35.191Z info: Registering GET /api/v1/user/:id
+  2021-10-05T07:32:35.192Z info: Registering GET /api/v1/echo
+  2021-10-05T07:32:35.192Z info: Registering POST /api/v1/echo
+  2021-10-05T07:32:35.192Z info: Registering POST /api/v1/post-check
+  2021-10-05T07:32:35.193Z info: Registering POST /api/v1/check-in
+  ```
+4. At this point, you can use any **HTTP client** to query one of these endpoints (i.e., `curl localhost:5000/api/v1/release-notes`). It should return a mock for release notes. On top of this, you can add the option to connect the logger, and there you go.
```
Comment (LoryApiFortress) [inline docs/api-testing/mark3/mocking.md:66]: 4. At this point, you can use any **HTTP client** to query one of these endpoints (i.e., `curl localhost:5000/api/v1/release-notes`: it should return a mock for release notes).

File: docs/api-testing/mark3/mocking.md [hunk 4]
```diff
@@ -0,0 +1,271 @@
...
+
+
+
+
+## E2E Mode
+When Piestry is run with `--e2e`, it will turn into a reverse proxy gateway and forward the requests based to the origin, according to the OpenAPI specification. The requirement is the "server" definition of the OpenAPI spec should lead to an actual location.
+
+In this mode, you can enable contract validators as well as capture mode.
+
+### Contract Validators
```
Comment (LoryApiFortress) [inline docs/api-testing/mark3/mocking.md:242]: Add
"### Contract Validators
 There are two types of validations you can activate, focusing on different areas."
 
also between line 148 and 150

File: docs/api-testing/mark3/mocking.md [hunk 5]
```diff
@@ -0,0 +1,271 @@
...
+There are two types of validations you can activate, focusing on different areas.
+
```
Comment (LoryApiFortress) [inline docs/api-testing/mark3/mocking.md:244]: Add here
"### Validate Request
 If you want to make sure your requests are compliant with the schema, Piestry can help you.
 Run it with the `--validate-request` switch to activate the validation of inbound requests."

File: docs/api-testing/mark3/mocking.md [hunk 6]
```diff
@@ -0,0 +1,271 @@
...
+### Capture Mode
+Capture mode is activated by passing the `--capture` parameter, followed by the path to a directory. As the requests go through, Piestry will capture the responses coming from the origin and save them to file.
+
+When `--capture` is executed without `--e2e`, Pietry will try to map the saved files to the OpenAPI definition and serve them as examples.
+
+
+
+## Logger
+
+PIestry can log full API conversations, including validation results, to a file or to the Sauce API Testing platform.
```
Comment (LoryApiFortress) [inline docs/api-testing/mark3/mocking.md:257]: Typo: Piestry

File: docs/api-testing/mark3/mocking.md [hunk 7]
```diff
@@ -0,0 +1,271 @@
...
+To activate the logger, use the `--logger` switch followed either by the path to a file, or a URL.
+
+In Sauce Labs, you can feed the data to the Sauce Labs API Logging, the first tool in a upcoming larger suite of API debugging tools.
+
+Sauce Labs API Logging
+1. Access your Sauce Labs API testing account.
+1. Create a webhook for a project.
+1. Compose the logger url as in
+   ```bash
+   https://{user}:{key}@app.saucelabs.com/api-testing/rest/v4/{hook_id}/logger
```
Comment (LoryApiFortress) [inline docs/api-testing/mark3/mocking.md:268]: https://{user}:{key}@api.us-west-1.saucelabs.com/api-testing/rest/v4/{hook_id}/logger

File: docs/api-testing/mark3/mocking.md [hunk 8]
```diff
@@ -0,0 +1,271 @@
...
+The validate response is similar to the ["validate examples" (mocking mode)](/api-testing/mark3/mocking/#validate-examples); the difference is that will validate the actual responses in an end-to-end session. Use the switch `--validate-response` to enable it.
+
+### Capture Mode
+Capture mode is activated by passing the `--capture` parameter, followed by the path to a directory. As the requests go through, Piestry will capture the responses coming from the origin and save them to file.
+
+When `--capture` is executed without `--e2e`, Pietry will try to map the saved files to the OpenAPI definition and serve them as examples.
+
+
+
+## Logger
```
Comment (LoryApiFortress) [inline docs/api-testing/mark3/mocking.md:255]: This section is ok as "hint" for the user but we need a separate page for the Logger

File: docs/api-testing/mark3/mocking.md [hunk 9]
```diff
@@ -0,0 +1,271 @@
...
+
+Sauce Labs API Logging
+1. Access your Sauce Labs API testing account.
+1. Create a webhook for a project.
+1. Compose the logger url as in
+   ```bash
+   https://{user}:{key}@app.saucelabs.com/api-testing/rest/v4/{hook_id}/logger
+   ```
+1. Launch Piestry with the given URL.
+1. Use the "Logger" section in the dashboard to inspect your calls, convert them into HTTP calls in the HTTP Client, or generate a test.
```
Comment (LoryApiFortress) [inline docs/api-testing/mark3/mocking.md:271]: this is not a sorted list

File: docs/api-testing/mark3/mocking.md (hunk 8 above)
Comment (kimsauce) [reply docs/api-testing/mark3/mocking.md:255]: done!

File: docs/api-testing/mocking.md [hunk 10]
```diff
@@ -0,0 +1,258 @@
...
+### Validate Request
+If you want to make sure your requests are compliant with the origin, run it with the `--validate-request` switch to activate the validation of inbound requests."
+
+### Validate Response
+This is similar to [**Validate examples** (mocking mode)](#validate-examples); the difference is that will validate the actual responses in an end-to-end session. Use the switch `--validate-response` to enable it.
+
+### Capture Mode
+Capture mode is activated by passing the `--capture` parameter, followed by the path to a directory. As the requests go through, Piestry will capture the responses coming from the origin and save them to file.
+
+When `--capture` is executed without `--e2e`, Pietry will try to map the saved files to the OpenAPI definition and serve them as examples.
```
Comment (LoryApiFortress) [inline docs/api-testing/mocking.md:253]: typo: replace "Pietry" with "Piestry"

File: docs/api-testing/logger.md [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+
+## What You'll Need
+* A Sauce Labs account ([Log in](https://accounts.saucelabs.com/am/XUI/#login/) or sign up for a [free trial license](https://saucelabs.com/sign-up)).
+
+
+## Running the Logger
+1. From your Sauce Labs API Testing account, go to a Project.
+2. Create a webhook for that project by clicking **Webhooks** > **Create hook**.
+3. Copy Hook URL and add `/logger` at the end as in:
+`https://{user}:{key}@api.us-west-1.saucelabs.com/api-testing/rest/v4/{hook_id}/logger`.
```
[the remark this answers is not in the document]
Comment (kimsauce) [reply docs/api-testing/logger.md:19]: @LoryApiFortress that's a good question. How about this? 

https://{SAUCE_USERNAME}:{SAUCE_ACCESS_KEY}@{SAUCE_REST_URL}/{hook_id}/logger

File: docs/api-testing/logger.md [hunk 12]
```diff
@@ -0,0 +1,43 @@
...
+4. From your CLI, launch our API Mocking tool, along with the `--logger` switch, followed by the URL mentioned in the previous step:
+   ```bash
+  docker run -v "$(pwd)/myspec:/specs" \
+  -p 5000:5000 quay.io/saucelabs/piestry \
+  -u /specs/myspec.yaml \
+  --logger https://{user}:{key}@api.us-west-1.saucelabs.com/api-testing/rest/v4/{hook_id}/logger
```
[the remark this answers is not in the document]
Comment (kimsauce) [reply docs/api-testing/logger.md:25]: @LoryApiFortress that's a good question. How about this?

https://{SAUCE_USERNAME}:{SAUCE_ACCESS_KEY}@{SAUCE_REST_URL}/{hook_id}/logger

File: docs/api-testing/mocking.md [hunk 13]
```diff
@@ -0,0 +1,254 @@
...
+* Isolating a microservice from the rest of the API actions so that everything else is stable and you can drill down to find the error.
+
+## What You'll Need
+
+* A Sauce Labs account ([Log in](https://accounts.saucelabs.com/am/XUI/#login/) or sign up for a [free trial license](https://saucelabs.com/sign-up))
+* An OpenAPI spec file.
+
+## Getting Started
+
+Piestry must be started from a Docker container in your CI/CI pipeline using Docker image, `quay.io/saucelabs/piestry`. Use the code snippet below, where `/specs/myspec.yaml` is the URI to your YAML spec file (can be local or remote):
```
Comment (LoryApiFortress) [inline docs/api-testing/mocking.md:30]: Typo: replace "CI/CI" with "CI/CD"

**Pull request merged**